graphql/tests/query.rs

```diff
@@ -1454,3 +1469,29 @@ async fn block_extension() {
         assert_eq!(None, result.extensions);
     }
 }
+
+#[tokio::test]
+async fn query_detects_reorg() {
+    let query = "query { musician(id: \"m1\") { id } }";
+    let query = graphql_parser::parse_query(query).expect("invalid test query");
+    let state = STORE
+        .deployment_state_from_id(TEST_SUBGRAPH_ID.clone())
+        .expect("failed to get state");
+    let result = execute_query_document_with_state(query.clone(), state);
+
+    assert!(result.errors.is_none());
+    assert_eq!(result.data.unwrap(), object!(musician: object!(id: "m1")));
+
+    // Simulate a reorg; this is what store::metadata::revert_block_ptr does
+    // though it goes through the database
+    let mut state = STORE
+        .deployment_state_from_id(TEST_SUBGRAPH_ID.clone())
+        .expect("failed to get state");
+    state.reorg_count += 1;
+    state.max_reorg_depth += 1;
+    let result = execute_query_document_with_state(query.clone(), state);
```

Comment:
So what will happen in this test is that a revert will be detected because the `state` reorg count which is `1` will be different from the store reorg count (which I assume is `0`). But in the real code isn't it an invariant that the query reorg count is always less or equal to the reorg count in the DB? If so that should ideally be respected in this test.

Reply:
Very good point! Fixing this, and having a true test that does a revert and forward move turned out to be pretty involved, since the tests in this file were all readonly and lazily set up the database. WIth a test that mutates, that's not possible any more. I've changed how the tests do their thing, forced them to be sequential, and changed the reorg test to go through reverting and moving forward again.